actions/download_samplesheet_clarity.py

```diff
@@ -0,0 +1,53 @@
+from st2actions.runners.pythonrunner import Action
+
+from clarity_ext.context import ExtensionContext
+from genologics.lims import *
+from genologics.config import BASEURI, USERNAME, PASSWORD
+
+import os
+from tempfile import mkdtemp
+
+class DownloadSamplesheetClarity(Action):
+    """
+    Downloads a samplesheet from Clarity LIMS (instead of Hermes)
+    keyed on the flowcell id.
+    """
+
+    def get_processes(self, containers):
+        for container in containers:
+            placement = container.get_placements()
+            artifact = placement[placement.keys()[0]]
+            yield artifact.parent_process.id
+
+    def highest_id(self, process_ids):
+        highest_proc_number = 0
+        highest_proc_type = ''
+        for id in process_ids:
+            (proc_type, proc_number) = id.split('-')
+            if highest_proc_number == 0 or int(proc_number) > highest_proc_number:
+                highest_proc_number = int(proc_number)
+                highest_proc_type = proc_type
+        return highest_proc_type + '-' + str(highest_proc_number)
+
+
+    def run(self, flowcell_name):
+        lims = Lims(BASEURI, USERNAME, PASSWORD) #Need credentials on stackstorm server for genologics package.
+        containers = lims.get_containers(name=flowcell_name)
+        processes = list(self.get_processes(containers))
+        newest_process_id = self.highest_id(processes)
+        context = ExtensionContext.create(newest_process_id)
+        cwd = os.getcwd()
```

Comment:
This is currently wrapped in clarity_ext code using API calls writing to disk, so no quick fix available.